scripts/lib/get-routes.js

```diff
@@ -39,23 +39,17 @@ function getRoutes () {
 
     // handle some exceptions. TODO: move this into @octokit/routes
     endpoints.forEach(endpoint => {
-      // exception for uploadReleaseAssets which passes parameters as header values
-      // see https://github.com/octokit/rest.js/pull/1043
-      if (endpoint.idName === 'uploadReleaseAsset') {
-        const contentLengthParam = endpoint.params.find(param => param.name === 'Content-Length')
-        const contentTypeParam = endpoint.params.find(param => param.name === 'Content-Type')
-        const indexes = endpoint.params.reduce((result, param, i) => {
-          if (['Content-Length', 'Content-Type'].includes(param.name)) {
-            result.unshift(i)
-          }
+      const headerParams = endpoint.params.filter(param => param.location === 'headers')
+      if (headerParams.length) {
+        headerParams.forEach(param => Object.assign(param, { name: `headers.${param.name.toLowerCase()}` }))
 
-          return result
-        }, [])
-        indexes.forEach(i => endpoint.params.splice(i, 1))
```

Comment:
I used the `indexes` variable so that the parameters remain in the same place. That is relevant for the docs that are also generated from this script.

Reply:
Ah 🤦‍♂️ now I know why you removed that. Okay in that case I think I’d go with #1119. If there will be more than one endpoints in future we can  still make the implementation more generic. It’s a problem when it’s a problem :)